scheduler/src/cook/mesos/constraints.clj

```diff
@@ -114,6 +114,18 @@
   (let [needs-gpus? (job-needs-gpus? job)]
     (->gpu-host-constraint job needs-gpus?)))
 
+(defrecord rebalancer-reservation-constraint [reserved-hosts]
```

Comment:
Should this be `host->job-ids` so we can add which jobs the host is reserved for in the message? If so, we should change `job-uuid->reserved-host` to `reserved-host->job-uuids` in the reservation atom.

Reply:
Missing unit tests for `rebalancer-reservation-constraint`.

Reply:
Should we consider passing in the raw `job-uuid->reserved-hostname` here (instead of building `reserved-hosts` in `build-rebalancer-reservation-constraint`) and then being smarter in `job-constraint-evaluate` while computing the return value? It will save us having to create the `reserved-hosts` for every job.

Reply:
I don't want to add the hostname to the string because we've had issues in this code before allocating tons of string objects for failure messages which are ultimately unused.

How were you thinking of changing the implementation? I was trying to front load the cost on constructing the constraint as opposed to the matching, which could happen much more often.

Added unit test.

Reply:
>  I was trying to front load the cost on constructing the constraint as opposed to the matching, which could happen much more often.

If we had `hostname->job-uuids`, we could use the same one for all jobs. Inside the job we would just check if the hostname entry exists and ensure the job is in the set of job-uuids for a given host.